src/Engine.cpp

```diff
@@ -321,15 +321,22 @@ bool Engine::fill_entity_info_(
         info_model_->update(default_info);
         info_model_->update_selected_entity(
             backend::entity_kind_to_QString(backend::EntityKind::INVALID),
-            "No entity selected");
+            "No entity selected",
+            "UNKNOWN_APP");
     }
     else
     {
         EntityInfo entity_info = backend_connection_.get_info(id);
+        std::string app_id = "UNKNOWN_APP";
+        if (entity_info.contains("app_id"))
+        {
+            app_id = entity_info["app_id"];
```

Comment:
Here I would check if the provided app_id is contained in Backend's recognized ids (include/types/app_names.h). Otherwise always set "UNKNOWN_APP".
Otherwise, it will try to find a non-existent file and throw an unnecessary error.